lib/std/base64.zig

```diff
@@ -1,3 +1,34 @@
+//! This module provides functions for encoding binary data to
+//! printable ASCII characters and decoding such encodings back to
+//! binary data.
+//!
+//! The encodings are described in RFC 4648.
+//!
+//! There are four variants defined here:
+//!
+//! standard:
+//!   Standard Base64 codecs, with padding
+//!
+//! standard_no_pad:
+//!   Standard Base64 codecs, without padding
+//!
+//! url_safe:
+//!   URL-safe Base64 codecs, with padding
+//!
+//! url_safe_no_pad:
+//!   URL-safe Base64 codecs, without padding
+//!
+//! Example:
```

Comment:
we have fn doc for this.
The example can reference a specific function with
```zig
test fnname {
    coolexample
    ...
}

Code should be always tested, if possible. No dead code.

Reply:
Roger. That is a good policy. I saw that there was talk of the roughness of the docs keeping the newbs out before stuff is ready for them. I started writing these docs to provide easy onboarding for new users. Maybe we don't want that yet?

Reply:
> I started writing these docs to provide easy onboarding for new users. Maybe we don't want that yet?

Comments have a specific purpose, ie top level docs giving an overview if it is feasible and meaningful.
Personally I prefer to keep top level docs very short with text descriptions, but this can be often seen as a taste thing.